x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/steps/blocks/action/math/math_expression_editor.tsx

```diff
@@ -0,0 +1,295 @@
+      const markers: monaco.editor.IMarkerData[] = validationResult.errors.map((error) => ({
+        severity: monaco.MarkerSeverity.Error,
+        message: error,
+        startLineNumber: 1,
+        startColumn: 1,
+        endLineNumber: model.getLineCount(),
+        endColumn: model.getLineMaxColumn(model.getLineCount()),
+      }));
+      monaco.editor.setModelMarkers(model, 'math-validation', markers);
+    } else {
+      // Clear markers when valid
+      monaco.editor.setModelMarkers(model, 'math-validation', []);
+    }
+  }, [hasInlineError, validationResult]);
+
+  // Register completion provider with field suggestions (re-register when suggestions change)
+  useEffect(() => {
+    const completionDisposable = registerMathCompletionProvider(fieldSuggestions);
+    return () => {
+      completionDisposable.dispose();
+    };
+  }, [fieldSuggestions]);
+
+  // Register signature help provider (static, only needs to be registered once)
+  useEffect(() => {
+    const signatureDisposable = registerMathSignatureHelpProvider();
+    return () => {
+      signatureDisposable.dispose();
+    };
+  }, []);
+
+  const handleWordWrapToggle = useCallback(() => {
+    const newValue = !isWordWrapped;
+    editorRef.current?.updateOptions?.({ wordWrap: newValue ? 'on' : 'off' });
+    setIsWordWrapped(newValue);
+  }, [isWordWrapped]);
+
+  // Format documentation sections for LanguageDocumentationPopover
+  const documentationSections = useMemo(() => {
+    const { groups, initialSection } = getMathExpressionLanguageDocSections();
+    return {
+      groups,
+      initialSection: <Markdown readOnly>{initialSection}</Markdown>,
+    };
+  }, []);
+
+  const containerStyles = css`
+    border: ${euiTheme.border.thin};
+    border-radius: ${euiTheme.border.radius.medium};
+    overflow: hidden;
+  `;
+
+  const headerStyles = css`
+    background-color: ${euiTheme.colors.backgroundBaseSubdued};
+    padding: ${euiTheme.size.xs} ${euiTheme.size.s};
+    border-bottom: ${euiTheme.border.thin};
+  `;
+
+  const footerStyles = css`
+    background-color: ${euiTheme.colors.backgroundBaseSubdued};
+    padding: ${euiTheme.size.xs} ${euiTheme.size.s};
+    border-top: ${euiTheme.border.thin};
+  `;
+
+  // Collect errors for display
+  const errors = hasInlineError ? validationResult.errors : [];
+
+  return (
+    <EuiFormRow
+      label={i18n.translate('xpack.streams.math.editorLabel', {
+        defaultMessage: 'Expression',
+      })}
+      isInvalid={errors.length > 0}
+      fullWidth
+    >
+      <div css={containerStyles} data-test-subj="streamsMathExpressionEditor">
+        {/* Header with word wrap toggle */}
+        <div css={headerStyles}>
+          <EuiFlexGroup alignItems="center" gutterSize="m" responsive={false}>
+            <EuiFlexItem grow={false}>
+              <EuiToolTip
+                content={
+                  isWordWrapped
+                    ? i18n.translate('xpack.streams.math.editorDisableWordWrap', {
```

Comment:
`xpack.streams.math.editorDisableWordWrap` and `xpack.streams.math.editorEnableWordWrap` could be moved to variables as they're both repeated in this file.